### ISSUE TITLE
Switch to using a Mock API

### ISSUE THREAD

**Issue by Monstarrrr** [Monstarrrr/rebutify#296]
## Problem
- Blocking backend from creating response schemas (to match with frontend expectations).  
- The front-end is dependent on the back-end during development leading to issues being blocked.

## Solution
Using a mock API  

----
❌❌❌❌❌❌❌❌❌❌❌❌❌❌❌❌ 
 ~~Re-create all existing API routes as new wiremock API routes (based on white paper API docs)~~

### ~~Challenge 1~~
~~*Using wiremock's authentication solution for their mock API isn't free, and it possibly implies implementing a different codebase due to differences in what schema it expects and returns.*~~
### ~~Solution~~
~~- Using the ``jwt``and ``jwks`` response template (https://docs.wiremock.io/response-templating/jwt/)~~
**Update**: It's annoying, the cURL commands don't allow me to access ``/__admin`` because "bad hostname" while it's the same command as in the docs. Then I get issues with trying to run the wiremock standalone java version to test it there because it's incompatible with the latest java version *of the runtime* specifically (55), then java archives ask me to create an account to download the executable ? I keep running into issues like these JUST to test ONE admin endpoint from wiremock.  
**TLDR; Let's switch to a less annoying environment and just use Express. We could perfectly match the existing authentication process without compromise.**  

### ~~Challenge 2~~  
~~*How to handle confirmation email upon registration ?*~~
### ~~Solution~~
~~- No straightforward solution with wiremock, gotta bypass the email and manually validating the account by retrieving the token from the registration endpoint even though in production that endpoint should not send a token back...~~
❌❌❌❌❌❌❌❌❌❌❌❌❌❌❌❌  

----  

### Challenge 1  
How to chose the schema for endpoints ?  
### Solution  
Endpoints don't have multiple methods to them (get/post/put/delete), instead they all serve on single method (e.g. GET arguments/{id} to get an argument; POST arguments/{id}/delete to delete an argument (and not [POST || DELETE] arguments/{id})

### Challenge 2  
We want to use a consistent approach to API endpoints and schemas for consistency (better maintainability) and for generating documentation automatically  
### Solution  
Following OpenAPI specs https://spec.openapis.org/oas/latest.html

### Challenge 3
*How to handle routes with more complex back-end logic like search routes and pagination ?*   
### Solution  
- We deliver random responses even if they don't match with the search or the requested page. This is because as a mock API, the goal is to deliver data with as little back-end logic as possible to save time, and not having the response match with the search request is acceptable since we are not testing the API, we are testing the front-end.  

----
- [x] #288 
- [x] Recreate existing routes as mock endpoints in ExpressJS 
  - [x] #293 
  - [x] #294   
- [x] Create a mock database
- [x] #310

  - ~~https://youtu.be/j6ow-UemzBc&t=431~~ good but about scaling 
  - https://youtu.be/5aryMKiBEKY
